src/sentry/profiles/task.py

```diff
@@ -1115,3 +1121,14 @@ def _calculate_duration_for_sample_format_v2(profile: Profile) -> int:
 
 def _calculate_duration_for_android_format(profile: Profile) -> int:
     return int(profile["duration_ns"] * 1e-6)
+
+
+def _set_frames_platform(profile: Profile):
+    if "version" in profile:
+        platform = profile["platform"]
+        for i, f in enumerate(profile["profile"]["frames"]):
+            if f.get("platform", "") == "":
+                profile["profile"]["frames"][i]["platform"] = platform
+    elif profile["platform"] == "android":
+        for i, _ in enumerate(profile["profile"]["methods"]):
+            profile["profile"]["methods"][i]["platform"] = "android"
```

Comment:
I think we could simplify here. Only the `android` case is special where we get frames from the `methods` key and this get rid of 2 loops relying on indices to modify a list, which is not needed and more error-prone.

```suggestion
    platform = profile["platform"]
    frames = profile["profile"]["methods"] if platform == "android" else profile["profile"]["frames"]
    for f in frames:
        if "platform" not in f:
            f["platform"] = platform
```